rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/KryoSerializerFactory.java

```diff
@@ -28,22 +28,50 @@
 import com.esotericsoftware.kryo.Serializer;
 import com.esotericsoftware.kryo.io.Input;
 import com.esotericsoftware.kryo.io.Output;
-import com.esotericsoftware.kryo.pool.KryoFactory;
-import com.esotericsoftware.kryo.pool.KryoPool;
+import com.esotericsoftware.kryo.util.Pool;
 import de.javakaffee.kryoserializers.JdkProxySerializer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * @author jsbxyyx
  */
-public class KryoSerializerFactory implements KryoFactory {
+public class KryoSerializerFactory {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(KryoSerializerFactory.class);
 
     private static final KryoSerializerFactory FACTORY = new KryoSerializerFactory();
 
-    private KryoPool pool = new KryoPool.Builder(this).softReferences().build();
+    private Pool<Kryo> pool = new Pool<Kryo>(true, true) {
```

Comment:
Why give up before use?

Reply:
?  I don't understand what you mean

Reply:
why change `private KryoPool pool = new KryoPool.Builder(this).softReferences().build();`  to `private Pool<Kryo> pool = new Pool<Kryo>(true, true)` ?

Reply:
> why change `private KryoPool pool = new KryoPool.Builder(this).softReferences().build();` to `private Pool<Kryo> pool = new Pool<Kryo>(true, true)` ?

Because 5. X KryoPool was deleted

Reply:
> why change `private KryoPool pool = new KryoPool.Builder(this).softReferences().build();` to `private Pool<Kryo> pool = new Pool<Kryo>(true, true)` ?

![image](https://user-images.githubusercontent.com/31852897/192773613-83931d99-4eed-4f5c-a5e9-230a15eb81c8.png)
Replaced by this class

Reply:
KryoFactory#create method does not exist in the new version?

Reply:
@jsbxyyx please take a look.

Reply:
@slievrly yes, 5.3.0 deleted builder api.